src/utils/forms/FormFactory.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {
+    FormControl,
+    FormGroup,
+    Col,
+    ControlLabel,
+    HelpBlock,
+    ButtonGroup,
+    DropdownButton,
+    MenuItem,
+    Checkbox
+  } from 'patternfly-react';
+
+export const FormFactory = (props) => {
+  return Object.keys(props.fields).map(key => {
+    if (props.fields[key].isVisible && props.fields[key].isVisible(props.fieldsValues) === false){
+      return undefined;
+    }
+    let child;
+    const validMsg = _.get(props.fieldsValues[key], 'validMsg', null);
+    switch (props.fields[key].type) {
+      case 'textarea':
+        child = <FormControl 
+                  componentClass="textarea"
+                  value={_.get(props.fieldsValues[key], 'value', '')}
+                  onChange={event => props.onFormChange(event.target.value, key)}
+                />;
+        break;
+      case 'dropdown':
+        child = <ButtonGroup justified>
+                  <DropdownButton
+                    id={`wizard-dropdown-${key}`}
+                    bsStyle="default"
+                    className="wizard-dropdown"
+                    title={_.has(props.fieldsValues[key], 'value') ? props.fieldsValues[key].value : props.fields[key].default}
+                    onSelect={value => props.onFormChange(value, key)}>
+                    {props.fields[key].values.map(value => <MenuItem key={value.id ? value.id : value.name} eventKey={value.id ? value.id : value.name}>{value.name}</MenuItem>)}
+                  </DropdownButton>
+                </ButtonGroup>;
+        break;
+      case 'checkbox':
+        child = <Checkbox 
+                  checked={_.get(props.fieldsValues[key], 'value', false)}
+                  onChange={event => props.onFormChange(event.target.checked, key)}
+                >
+                 {props.fields[key].title}
+               </Checkbox>;
+        break;
+      default:
+        child = <FormControl 
+                  type="text"
+                  value={_.get(props.fieldsValues[key], 'value', '')}
+                  onChange={event => props.onFormChange(event.target.value, key)}
+                />;
+    }
+    return (
+        <FormGroup key={key} validationState={validMsg ? 'error' : null}>
+          <Col sm={_.get(props.fields[key], 'labelSize') ? props.fields[key].labelSize : 3} className="text-right">
+            {props.fields[key].type === 'checkbox' ? null:
+              <ControlLabel className={props.fields[key].required ? 'required-pf': null}>{props.fields[key].title}</ControlLabel>}
+          </Col>
+          <Col sm={_.get(props.fields[key], 'labelSize') ? 8-props.fields[key].labelSize : 5}>
+            {child}
+            <HelpBlock>{validMsg ? validMsg: null}</HelpBlock>
+          </Col>
+        </FormGroup>);
+  })
+}
```

Comment:
newline - better to setup an IDE to do it for you ..

Reply:
fixed